packages/flutter_tools/lib/src/version.dart

```diff
@@ -32,9 +32,10 @@ class FlutterVersion {
     return _repositoryUrl;
   }
 
-  /// Whether we are currently on the stable branch.
-  bool get isStable {
-    return getBranchName() == 'stable';
+  /// Whether we are currently on the master branch.
+  bool get isMaster {
+    final String branchName = getBranchName();
+    return !<String>['dev', 'beta', 'stable'].contains(branchName);
```

Comment:
This feels like it will fail for people who are working on a git branch.

Reply:
Unless we also want to stop all flutter engineers from working it, we'll need to permit user branches somehow